src/main/java/com/minecolonies/coremod/commands/GetColonyAndCitizen.java

```diff
@@ -0,0 +1,196 @@
+            }
+            catch (NumberFormatException e)
+            {
+                citizenId = -1;
+            }
+        }
+        else if (args.length == 3 && colonyId >= 0)
+        {
+            citizenName = args[0] + " " + args[1] + " " + args[2];
+            for (int i = 1
+                   ; i <= ColonyManager.getColony(colonyId).getCitizens().size(); i++)
+            {
+                try
+                {
+                    if (ColonyManager.getColony(colonyId).getCitizen(i).getName().equals(citizenName))
+                    {
+                        citizenId = i;
+                    }
+                }
+                catch (NullPointerException e)
+                {
+                    continue;
+                }
+            }
+
+        }
+        else if (args.length == 4 && colonyId >= 0)
+        {
+            citizenName = args[1] + " " + args[2] + " " + args[3];
+            for (int i = 1; i <= ColonyManager.getColony(colonyId).getCitizens().size(); i++)
+            {
+                if (ColonyManager.getColony(colonyId).getCitizen(i).getName().equals(citizenName))
+                {
+                    citizenId = i;
+                }
+            }
+        }
+        else if (args.length >= 5)
+        {
+            citizenId = -2;
+        }
+        else
+        {
+            citizenId = -3;
+        }
+        if (citizenId >= 0 && colonyId >= 0)
+        {
+            if (ColonyManager.getColony(colonyId).getCitizen(citizenId) == null)
+            {
+                citizenId = -4;
+            }
+        }
+        return citizenId;
+    }
+
+    public static String getErrors(@NotNull final int colonyId, @NotNull final int citizenId)
+    {
+        String ErrorMessage = null;
+        if (colonyId >= 0)
+        {
+            ErrorMessage = null;
+        }
+        else if (colonyId == -1)
+        {
+            ErrorMessage = String.format(ONLY_NUMBERS, "colony");
+        }
+        else if (colonyId == -2)
+        {
+            ErrorMessage = TOO_MANY_ARGUMENTS;
+        }
+        else if (colonyId == -3)
+        {
+            ErrorMessage = UNKNOWN_ERROR;
+        }
+        else if (colonyId == -4)
+        {
+            ErrorMessage = String.format(NOT_FOUND, "Colony");
+        }
+        else if (colonyId == -5)
+        {
+            ErrorMessage = NO_COLONY;
+        }
+
+        if (citizenId >= 0 && ErrorMessage == null)
+        {
+            ErrorMessage = null;
+        }
+        else if (citizenId == -1 && ErrorMessage == null)
+        {
+            ErrorMessage = String.format(ONLY_NUMBERS, "citizen");
+        }
+        else if (citizenId == -2 && ErrorMessage == null)
+        {
+            ErrorMessage = TOO_MANY_ARGUMENTS;
+        }
+        else if (citizenId == -3 && ErrorMessage == null)
+        {
+            ErrorMessage = UNKNOWN_ERROR;
+        }
+        else if (citizenId == -4 && ErrorMessage == null)
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 4 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)